adam-core/src/main/scala/org/bdgenomics/adam/hbase/HBaseFunctions.scala

```diff
@@ -0,0 +1,587 @@
+import org.apache.hadoop.hbase.client.Admin
+import org.apache.hadoop.hbase.client.ConnectionFactory
+import org.apache.hadoop.hbase.client.Delete
+import org.apache.hadoop.hbase.client.Put
+import org.apache.hadoop.hbase.client.Scan
+import org.apache.hadoop.hbase.client.Get
+import org.apache.hadoop.hbase.client.Table
+import org.apache.hadoop.hbase.client.Result
+import org.apache.hadoop.hbase._
+import org.apache.hadoop.hbase.spark._
+import org.apache.hadoop.hbase.spark.HBaseRDDFunctions._
+import org.apache.hadoop.hbase.util.Bytes
+import org.apache.hadoop.hbase.{ HBaseConfiguration, TableName }
+import org.apache.hadoop.hbase.io.compress.Compression.Algorithm
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.hbase.mapreduce.LoadIncrementalHFiles
+import org.apache.hadoop.fs.{ FSDataInputStream, FileSystem, Path }
+import org.apache.spark.SparkContext
+import org.apache.spark.rdd.RDD
+import org.bdgenomics.adam.rdd.variant.{ VariantContextRDD }
+import org.bdgenomics.adam.rich.RichVariant
+import org.bdgenomics.formats.avro._
+import org.bdgenomics.adam.models.{ ReferencePosition, ReferenceRegion, SequenceDictionary, VariantContext }
+import scala.io.Source
+import scala.collection.mutable.ListBuffer
+import scala.collection.JavaConverters._
+import sys.process._
+
+object HBaseFunctions {
+
+  /**
+   * HBase Spark Data Access Object
+   *
+   * In standard usage user will create a single instance of this object, and pass it as the first parameter
+   * when using the HBaseFunctions public API functions
+   *
+   * @param sc SparkContext
+   * @param inConf optional HBase Configuration parameter, only used if manually configuring HBase connection
+   * @param inConnection optional HBase Connection parameter, only used if manually configuring HBase connection
+   * @param inAdmin optional HBase Admin parameter, only used if manually configuring HBase connection
+   * @param inHBaseContext optional HBaseContext parameter, only used if manually configuring HBase connection
+   */
+
+  class HBaseSparkDAO(sc: SparkContext,
+                      inConf: Option[Configuration] = None,
+                      inConnection: Option[Connection] = None,
+                      inAdmin: Option[Admin] = None,
+                      inHBaseContext: Option[HBaseContext] = None) {
+
+    val conf = if (inConf.isDefined) inConf.get else HBaseConfiguration.create()
+    val connection = if (inConnection.isDefined) inConnection.get else ConnectionFactory.createConnection(conf)
+    val admin = if (inAdmin.isDefined) inAdmin.get else connection.getAdmin
+    val hbaseContext = if (inHBaseContext.isDefined) inHBaseContext.get else new HBaseContext(sc, conf)
+
+    def createTable(tableDescriptorMeta: HTableDescriptor) { admin.createTable(tableDescriptorMeta) }
+    def getTable(tableName: TableName): Table = { connection.getTable(tableName) }
+
+    def getHBaseRDD(tableName: org.apache.hadoop.hbase.TableName,
```

Comment:
I feel like this should be package private. I feel that an RDD of non-decoded results shouldn't be exposed as a public API.